### PR TITLE
Compile only tests

### PR DESCRIPTION
**Pull request by hkaiser**
Simplify the creation of compile-only tests in the build system

### CODE REVIEW

File: tests/headers/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2015 Hartmut Kaiser
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# collect a list of all header files in the source tree
+file(GLOB_RECURSE headers "${PROJECT_SOURCE_DIR}/hpx/*hpp")
```
Comment (sithhell) [inline tests/headers/CMakeLists.txt]: We should probably exclude everything in directories in "detail" directories, those are not meant to be included by anyone knowing exactly what they do anyways.

Comment (hkaiser) [inline tests/headers/CMakeLists.txt]: Good point, will do.

**Comment by sithhell** [comment]
I think we should run those tests in circle-ci to catch those errors before hitting master.

**Comment by hkaiser** [comment]
> I think we should run those tests in circle-ci to catch those errors before hitting master.

Good point. Lot'sa work fixing those, though...

**Comment by sithhell** [comment]
I am going to merge this PR now. The only remaining issue is documented here: #1748

**Pull request merged**